core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -137,6 +137,7 @@
     public static final String ENABLE_LARGE_DYNAMIC_FILTERS = "enable_large_dynamic_filters";
```

Comment:
Because streams that are not partitioned will have an empty partitioning arguments array (returns true). Similarly, streams partitioned only on constant will return true.

In both of these cases, you can have fixed arbitrary distribution local exchange inserted into the stage plan to increase parallelism, followed by gathering exchanges after which the resulting stream is no longer sorted.